lib/play_game.py

```diff
@@ -10,20 +10,31 @@ def __init__(self, board, ui):
     def start(self):
         self.ui.display("Welcome to TIC TAC TOE!")
         self.board.display_board()
+        self.ui.display("X goes first")
+        current_player = "X"
         while not self.board.board_full(self.board.board):
-            self.turn()
+            self.turn(current_player)
+            current_player = self.next_player(current_player)
 
-    def turn(self):
+    def turn(self, mark):
         self.ui.display("Choose a number between 1 and 9:")
         input_number = input()
+
         if self.valid_input(input_number) and not self.board.position_taken(
             input_number
         ):
-            self.board.move(input_number, "X")
+            self.board.move(input_number, mark)
             self.board.display_board()
+            self.ui.display("It's " + self.next_player(mark) + "'s turn")
         else:
             self.ui.display("Invalid number")
-            self.turn()
+            self.turn(mark)
 
     def valid_input(self, input):
         return input in self.board.board.keys()
+
+    def next_player(self, mark):
+        if mark == "X":
+            return "O"
+        elif mark == "O":
+            return "X"
```

Comment:
You're definitely on the right path here. My only thought here is that this code is not easily extensible - in the future when we ask for custom markers, this logic will no longer work. To piggy back off of Malisa's comment regarding naming this to `switch_players`, you could have this class use variables to store the two players' markers and then have `switch_players` simply swap those values. Something along the lines of:

```python
def switch_players(self, mark):
    self.current_player, self.next_player = self.next_player, self.current_player
```

Reply:
No problem! You'll learn more about these code smells as you progress through your apprenticeship. Consider this as my one and only freebie 😉